components/Banner/FloatingBanner.js

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react'
+import { parseImgUrl } from 'utils/common'
+import Link from 'next/link'
+import ParasRequest from 'lib/ParasRequest'
+import useSWR from 'swr'
+
+const FloatingBanner = () => {
+	const fetchFloatingBanner = () =>
+		ParasRequest.get(`${process.env.V2_API_URL}/floating-banner`).then((res) => res.data.result[0])
+
+	const { data, isValidating } = useSWR('floating-banner', fetchFloatingBanner)
+	const [showFloating, setShowFloating] = useState(false)
+
+	useEffect(() => {
+		if (data?.is_active) {
+			setShowFloating(true)
+		} else {
+			setShowFloating(false)
+		}
+	}, [data])
+
+	if ((!data && isValidating) || data.length === 0) {
+		return null
+	}
+
+	return (
+		<>
+			{showFloating && (
+				<Link href={`${data?.open_link}`}>
```

Comment:
daripada `showFloating`nya di bikin state, sebenernya bisa langsung ambil dari data aja sih mestinya. concernnya buat ngurangin unused render aja sih, sama biar lebih enak bacanya

```js
const { data, isValidating } = useSWR('floating-banner', fetchFloatingBanner)

if ((!data && isValidating) || data.length === 0) {
  return null
}

return (
  <>
    {/* langsung gini */}
    {data.is_active && (
      <Link href={`${data?.open_link}`}>
```

kalo mau baca2 lebih lanjut [disini](https://beta.reactjs.org/learn/you-might-not-need-an-effect)